### PR TITLE
CI Update

### PR DESCRIPTION
**Pull request by CiaranOMara**
This pull request was created in response to https://github.com/GiovineItalia/Gadfly.jl/pull/1339#issuecomment-550114393.
# Proposed changes

- ~~Add Julia 1.3 to test matrix.~~
- Use the codecov flag in Travis instead of the manual code version.
- Pkg.instantiate after Pkg.develop.
- Add colour to make doc output.
- Move tests with Compose.jl#master into an Edge stage.

Moving tests using Compose.jl#master into their own stage provides visibility for new development while prioritising Gadfly.jl#master.

### CODE REVIEW

**Comment by CiaranOMara** [comment]
I wasn't sure how to best structure this pull request. Anyhow, I've added Cairo.jl compatibility on the back of this PR as it not only shows improvement across operating systems, it also shows how the Edge stage works - you'll notice that many of the deprecation warnings are already addressed by Compose.jl#master.

**Comment by Mattriks** [comment]
I assume this PR will close #1320 and #1324. If so, add that to your first post above (https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords)

**Comment by bjarthur** [comment]
it'd be better to have the Cairo dependency commit in a separate PR.  i'd merge that right away.

regarding 1.3, again, i'd suggest holding off on that until it is actually released, and stop testing on 1.2 at the same time.

i also don't understand the Edge job.  i'm just being dumb here i think.  can you please explain how this is different?

**Comment by CiaranOMara** [comment]
Previously, by way of the `before_script`, all tests were performed using `Compose.jl#master`. With this PR, tests in the **Test job** use a `Compose.jl` release and tests in the **Edge job** use `Compose.jl#master`. With this configuration, the tests in the **Test job** test the packages installed by the end-user, and the tests in the **Edge job** shows developers how `Gadfly.jl` is performing with the latest `Compose.jl` developments.

**Comment by CiaranOMara** [comment]
I have moved the Cairo compatibility change to https://github.com/GiovineItalia/Gadfly.jl/pull/1343.

**Comment by CiaranOMara** [comment]
I have removed Julia v1.3 from the test matrix.

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,43 @@
 language: julia
+
 os:
   - linux
+  - osx
+  - windows
```
Comment (bjarthur) [inline .travis.yml]: the thinking in only testing Gadfly on linux is that all the OS-specific code is in Compose.  no need to burden the Travis servers.

Comment (CiaranOMara) [reply .travis.yml]: I did think that to be the case. Though their brief inclusion was useful to see Gadfly working with the new binary provided Cairo. Anyhow, I've removed macOS and Windows from the test matrix.

**Comment by CiaranOMara** [comment]
The Edge stage does not seem quite right as it does not appear to be allowed to fail.

**Pull request closed without merging**